gcloud/datastore/test_query.py

```diff
@@ -200,224 +182,161 @@ def test_add_filter_w_known_operator_and_entity(self):
         self.assertEqual(query.filters, [('other', '=', other)])
 
     def test_add_filter_w_whitespace_property_name(self):
-        query = self._makeOne()
+        _DATASET = 'DATASET'
+        query = self._makeOne(_DATASET)
         PROPERTY_NAME = '  property with lots of space '
         query.add_filter(PROPERTY_NAME, '=', u'John')
         self.assertEqual(query.filters, [(PROPERTY_NAME, '=', u'John')])
 
     def test_add_filter___key__valid_key(self):
         from gcloud.datastore.key import Key
-        query = self._makeOne()
+        _DATASET = 'DATASET'
+        query = self._makeOne(_DATASET)
         key = Key('Foo', dataset_id='DATASET')
         query.add_filter('__key__', '=', key)
         self.assertEqual(query.filters, [('__key__', '=', key)])
 
     def test_filter___key__invalid_operator(self):
         from gcloud.datastore.key import Key
+        _DATASET = 'DATASET'
         key = Key('Foo', dataset_id='DATASET')
-        query = self._makeOne()
+        query = self._makeOne(_DATASET)
         self.assertRaises(ValueError, query.add_filter, '__key__', '<', key)
 
     def test_filter___key__invalid_value(self):
-        query = self._makeOne()
+        _DATASET = 'DATASET'
+        query = self._makeOne(_DATASET)
         self.assertRaises(ValueError, query.add_filter, '__key__', '=', None)
 
     def test_projection_setter_empty(self):
+        _DATASET = 'DATASET'
         _KIND = 'KIND'
-        query = self._makeOne(_KIND)
+        query = self._makeOne(_DATASET, _KIND)
         query.projection = []
         self.assertEqual(query.projection, [])
 
     def test_projection_setter_string(self):
+        _DATASET = 'DATASET'
         _KIND = 'KIND'
-        query = self._makeOne(_KIND)
+        query = self._makeOne(_DATASET, _KIND)
         query.projection = 'field1'
         self.assertEqual(query.projection, ['field1'])
 
     def test_projection_setter_non_empty(self):
+        _DATASET = 'DATASET'
         _KIND = 'KIND'
-        query = self._makeOne(_KIND)
+        query = self._makeOne(_DATASET, _KIND)
         query.projection = ['field1', 'field2']
         self.assertEqual(query.projection, ['field1', 'field2'])
 
     def test_projection_setter_multiple_calls(self):
+        _DATASET = 'DATASET'
         _KIND = 'KIND'
         _PROJECTION1 = ['field1', 'field2']
         _PROJECTION2 = ['field3']
-        query = self._makeOne(_KIND)
+        query = self._makeOne(_DATASET, _KIND)
         query.projection = _PROJECTION1
         self.assertEqual(query.projection, _PROJECTION1)
         query.projection = _PROJECTION2
         self.assertEqual(query.projection, _PROJECTION2)
 
     def test_keys_only(self):
+        _DATASET = 'DATASET'
         _KIND = 'KIND'
-        query = self._makeOne(_KIND)
+        query = self._makeOne(_DATASET, _KIND)
```

Comment:
Relying on positions of kwargs makes me uncomfortable.

Am I just paranoid? Do we just have different tastes here?

I do know that in a stable API, we couldn't even change the order of kwargs without breaking user code, so once we reach `1.0` we'd be stuck. (This is somewhat an argument for just using `**kwargs` and then popping from it.)